plugins/statistics.koplugin/main.lua

```diff
@@ -73,58 +86,429 @@ function ReaderStatistics:init()
+            if lfs.attributes(path, "mode") == "file" then
+                local old_data = self:importFromFile(statistics_dir, curr_file)
+                if old_data and old_data.total_time > 0 and not exclude_titles[old_data.title] then
+                    local book_stats = {}
+                    for _, v in pairs(old_data.details) do
+                        book_stats.performance_in_pages[v.time] = v.page
+                    end
+                    book_stats.title = old_data.title
+                    book_stats.authors = old_data.authors
+                    book_stats.notes = old_data.notes
+                    book_stats.highlights = old_data.highlights
+                    book_stats.pages = old_data.pages
+                    book_stats.series = old_data.series
+                    book_stats.language = old_data.language
+                    book_stats.file = nil
+                    if self:addBookStatToDB(book_stats, conn) then
+                        nr_of_conv_books = nr_of_conv_books + 1
+                    else
+                        logger.dbg("Book not converted (old stats): " .. book_stats.title)
+                    end
+                end
+            end
+        end
+    end
+    return nr_of_conv_books
+end
+
+function ReaderStatistics:getIdBookDB()
+    local conn = SQ3.open(db_location)
+    local id_book
+    local book_title = self.data.title:gsub("'", "''")
+    local book_authors = self.data.authors:gsub("'", "''")
+    local sql_stmt = [[
+        SELECT count(id)
+        FROM   book
+        WHERE  title = '%s'
+            AND authors = '%s'
+            AND md5 = '%s'
+    ]]
+    local res = conn:rowexec(string.format(sql_stmt, book_title, book_authors, self.data.md5))
+    if res == 0 then
+        local stmt = conn:prepare("INSERT INTO book VALUES(NULL, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)")
+        stmt:reset():bind(self.data.title, self.data.authors, self.data.notes,
+            TimeVal:now().sec, self.data.highlights, self.data.pages,
+            self.data.series, self.data.language, self.data.md5, self.curr_total_time, self.curr_total_pages):step()
+        sql_stmt = [[
+            SELECT last_insert_rowid() AS num;
+        ]]
+        id_book = conn:rowexec(sql_stmt)
+    else
+        sql_stmt = [[
+            SELECT id
+            FROM   book
+            WHERE  title = '%s'
+                AND    authors = '%s'
+                AND    md5 = '%s'
+        ]]
+        id_book = conn:rowexec(string.format(sql_stmt, book_title, book_authors, self.data.md5))
+    end
+    conn:close()
+    return tonumber(id_book)
+end
+
+function ReaderStatistics:insertDB(id_book)
+    self.pages_stats[TimeVal:now().sec] = self.curr_page
+    if id_book == nil or util.tableSize(self.pages_stats) < 2 then
+        return
+    end
+    local diff_time
+    local conn = SQ3.open(db_location)
+    local sorted_performance = {}
+    for k, v in pairs(self.pages_stats) do
```

Comment:
when you are iterating through a table, it's better to give `k`, `v` specific names like `time` and `page`.